WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationCreatingFragment.java

```diff
@@ -214,10 +216,20 @@ private void mutateToCompleted(boolean showWebView) {
     }
 
     private PreviewWebViewClient loadWebview() {
-        String siteAddress = "https://" + getArguments().getString(ARG_SITE_SLUG) + ".wordpress.com";
-        PreviewWebViewClient client = new PreviewWebViewClient(siteAddress);
+        Assert.assertNotNull("Arguments can't be null at this point!", getArguments());
```

Comment:
Is it okay to use `Assert` here even though it's from the testing framework?  I haven't used it outside of tests so I'm not sure how it works in this case.  I tried forcing the assertion to fail by added `Assert.assertNotNull("Object is null!", null);` as the first statement in the `loadWebview()` method.  The following stack trace was in the log.  However, the app didn't crash and the progress spinner continued, which leaves the user waiting indefinitely.

```
2019-01-07 18:54:45.678 3937-3937/org.wordpress.android.beta E/EventBus: Could not dispatch event: class org.wordpress.android.ui.accounts.signup.SiteCreationService$SiteCreationState to subscribing class class org.wordpress.android.ui.accounts.signup.SiteCreationCreatingFragment
    junit.framework.AssertionFailedError: Object is null!
        at junit.framework.Assert.fail(Assert.java:50)
        at junit.framework.Assert.assertTrue(Assert.java:20)
        at junit.framework.Assert.assertNotNull(Assert.java:218)
        at org.wordpress.android.ui.accounts.signup.SiteCreationCreatingFragment.loadWebview(SiteCreationCreatingFragment.java:219)
        at org.wordpress.android.ui.accounts.signup.SiteCreationCreatingFragment.onSiteCreationStateUpdated(SiteCreationCreatingFragment.java:379)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:485)
        at org.greenrobot.eventbus.EventBus.postToSubscription(EventBus.java:420)
        at org.greenrobot.eventbus.EventBus.postSingleEventForEventType(EventBus.java:397)
        at org.greenrobot.eventbus.EventBus.postSingleEvent(EventBus.java:370)
        at org.greenrobot.eventbus.EventBus.post(EventBus.java:251)
        at org.greenrobot.eventbus.EventBus.postSticky(EventBus.java:292)
        at org.wordpress.android.util.AutoForeground.notifyState(AutoForeground.java:191)
        at org.wordpress.android.util.AutoForeground.setState(AutoForeground.java:169)
        at org.wordpress.android.ui.accounts.signup.SiteCreationService.setState(SiteCreationService.java:257)
        at org.wordpress.android.ui.accounts.signup.SiteCreationService.executePhase(SiteCreationService.java:371)
        at org.wordpress.android.ui.accounts.signup.SiteCreationService.finishedPhase(SiteCreationService.java:379)
        at org.wordpress.android.ui.accounts.signup.SiteCreationService.access$100(SiteCreationService.java:46)
        at org.wordpress.android.ui.accounts.signup.SiteCreationService$2.run(SiteCreationService.java:488)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
```

Since we have a fallback interface when `loadWebview()` doesn't work with the "tada" view, we can catch and log the `NullPointerException`s for `getArguments()` and `siteAddress` without hurting the user experience with a crash or indefinite interface.

Reply:
We do use tools like `requireNotNull` on Kotlin fairly often, but I agree using `Assert` in java does feel a bit weird. I can revert that change if you prefer as I have mostly added that to avoid the annoying AS warning. Kotlin crashes it in these cases which is what we want and I thought Assert would also crash here, but maybe not 🤷‍♂️

Reply:
I feel the same way about those lint warnings.  They aren't pretty, but here are a couple ways to avoid those warnings.

```java
    private PreviewWebViewClient loadWebview() {
        String siteAddress = getArguments() != null ? getArguments().getString(ARG_SITE_ADDRESS, "") : "";

        if (siteAddress.isEmpty()) {
            return null;
        }
```

```java
    private PreviewWebViewClient loadWebview() {
        if (getArguments() == null) {
            return null;
        }

        String siteAddress = getArguments().getString(ARG_SITE_ADDRESS);

        if (siteAddress == null) {
            return null;
        }
```

Reply:
We don't want to return `null` though, that's the thing. If we did, this is a non-problem, but that'd make the app crash in a different place.

It's not worth spending too much time on this, as it'll be completely removed very soon. But, I'll check it out tomorrow morning and improve it a bit. Thank you for the feedback!

Reply:
Returning `null` from the `loadWebview()` method will not make the app crash in a different place.  That's what I meant by this.

> Since we have a fallback interface when `loadWebview()` doesn't work with the "tada" view, we can catch and log the `NullPointerException`s for `getArguments()` and `siteAddress` without hurting the user experience with a crash or indefinite interface.

The `loadWebview()` method will assign `mPreviewWebViewClient` to `null` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/098e2cd2109454696cb23e58895e1407d848c06b/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationCreatingFragment.java#L370) during the `PRELOAD` step, which will eventually cause the only other instance of `loadWebview()` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/098e2cd2109454696cb23e58895e1407d848c06b/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationCreatingFragment.java#L376-L378) in the `SUCCESS` step.  Then `mutateToCompleted(mWebViewLoadedInTime)` will show the "tada" view since `mWebViewLoadedInTime` is `false`.

We can fail silently and just return `null` or we can catch and log the `NullPointerException`s for troubleshooting.  Either way we choose, avoiding a crash is a major user experience improvement.  I think it's worth spending time on it if it's going to be released.  Even if we replace it completely in the following release, we can't ensure users will update the app as quickly as we replaced it.

Reply:
You're right, I missed that. I think we have a different take on crashes within the team. We haven't discussed it very much on Slack or p2, but some devs (including myself) have been trying to let the app crash in invalid states, so that we can address the root cause of the said invalid state. So, we have been trying to do fail fast in a lot of different places. @malinajirka is one of the biggest advocates of this 😁 

In any case, I am going to see how we can improve this the best and ping you on it once I have something. Thanks again for the detailed feedback, really appreciate it! 🙇

Reply:
@theck13 I ended up going with your initial suggestion with just a small modification by throwing an `IllegalStateException` in `DEBUG` builds in 598e608a848fdd5c883d3d9b4dfea01de6c260f7. Let me know what you think!